lib/ansible/cli/galaxy.py

```diff
@@ -321,6 +489,61 @@ def execute_install(self):
         uses the args list of roles to be installed, unless -f was specified. The list of roles
         can be a name (which will be downloaded via the galaxy API and github), or it can be a local tar archive file.
         """
+        if context.CLIARGS['type'] == 'collection':
+            collections = context.CLIARGS['args']
+            force = context.CLIARGS['force']
+            output_path = context.CLIARGS['collections_path']
+            # TODO: use a list of server that have been configured in ~/.ansible_galaxy
+            servers = [context.CLIARGS['api_server']]
+            ignore_certs = context.CLIARGS['ignore_certs']
+            ignore_errors = context.CLIARGS['ignore_errors']
+            requirements_file = context.CLIARGS['requirements']
+            no_deps = context.CLIARGS['no_deps']
+            force_deps = context.CLIARGS['force_with_deps']
+
+            if collections and requirements_file:
+                raise AnsibleError("The positional collection_name arg and --requirements-file are mutually exclusive.")
+            elif not collections and not requirements_file:
+                raise AnsibleError("You must specify a collection name or a requirements file.")
+
+            if requirements_file:
+                requirements_file = os.path.expanduser(os.path.expandvars(requirements_file))
+                if not os.path.isabs(requirements_file):
+                    requirements_file = os.path.abspath(requirements_file)
+
+                collection_requirements = parse_collections_requirements_file(requirements_file)
+            else:
+                collection_requirements = []
+                for collection_input in collections:
+                    if ':' in collection_input:
```

Comment:
It's simpler like this (less branches + takes into account `collection_input == 'pkg_name:'`:
```suggestion
                    name, _sep, requirement = collection_input.partition(':')
					if not requirement:
                        requirement = '*'
```

Reply:
or even
It's simpler like this (less branches + takes into account `collection_input == 'pkg_name:'`:
```suggestion
                    name, _sep, requirement = collection_input.partition(':')
                    collection_requirements.append((name, requirement or '*', None))
```

Reply:
Thanks I've never knew about partition, looks a lot cleaner.